internal/filter/expr/matcher.go

```diff
@@ -63,3 +63,31 @@ func Or[K any](matchers ...BoolExpr[K]) BoolExpr[K] {
 		return orMatcher[K]{matchers: matchers}
 	}
 }
+
+type andMatcher[K any] struct {
```

Comment:
You could make this type like this:

```golang 
type andSliceMatchers [K any] []BoolExpr[K]

func Any[K any](matches ...BoolExpr[K]) BoolExpr[K] {
    return andSliceMatchers(matches) // Doesn't matter the length of matches since we iterator through each item regardless
}
```

Reply:
I was following the existing convention in this file with orMatcher, would you like me to refactor orMatcher as well?

Reply:
Lets, keeping the code simple makes it easier to read later on :)

Reply:
refactored the orMatcher and andMatcher. I wasn't able to do the same with the notMatcher because of type resolution issues I didn't understand. I think it's different because while the and and or matchers are pointers to arrays, the notMatcher is a singleton